### PR TITLE
Updating lints page to reflect recent changes

### LINKED ISSUE #102743: Future-Incompatible Lint - Improve Example
I was reading about future-incompatible lints on [this](https://doc.rust-lang.org/rustc/lints/index.html) page.

First of all, I struggled to reproduce the lint warning, and it would have been really educational if you provided an example of how to trigger that lint warning for future incompatibility (please add example of how I can trigger the warning showed in that section).

Second, I tried applying the `#![warn(safe_packed_borrows)]` in Rust Playground, like this:

```
#![allow(safe_packed_borrows)]
fn main() {}
```
Running the code produces, among other things, the following:
```
warning: lint `safe_packed_borrows` has been renamed to `unaligned_references`
```
So that means that you should update the "[example of what a future-incompatible looks like](https://doc.rust-lang.org/rustc/lints/index.html)", because the current example is obsolete.

### PR DESCRIPTION
**Pull request by llamaha**
Closes rust-lang/rust#102743

### CODE REVIEW

File: src/doc/rustc/src/lints/index.md [hunk 1]
```diff
@@ -39,19 +39,32 @@ provides detailed information and an opportunity for feedback. This gives
...
+}
+
+fn main() {
+    let packed = Packed { f1: 1, f2: 2 };
+    let _ = &packed.f2; // This line triggers the warning
+}
+
+$ rustc main.rs
+
+error[E0793]: reference to packed field is unaligned
```
Comment (Nilstrieb) [inline src/doc/rustc/src/lints/index.md:59]: that's not actually a future incompat lint now, it's just an error, you need a different example. search for `@future_incompat` in the compiler source to find examples

**Comment by ehuss** [comment]
I'm not entirely sure it is a good idea to try to change this to a new lint example. The set of future-incompatible lints is constantly changing, and in a few months or years this would likely just be back where it started. It's not clear to me what kind of concern the original issue poster was having, or why they wanted to recreate the output. For me, the existing text seems pretty clear that this is just an illustrative example of what it *might* look like. Can you say more about why it is important to have an example that is triggered in the current compiler?

File: src/doc/rustc/src/lints/index.md (hunk 1 above)
Comment (llamaha) [reply src/doc/rustc/src/lints/index.md]: Thanks @Nilstrieb , I learned a lot hunting these down.

**Comment by llamaha** [comment]
@ehuss I submitted a change before reading your comment.

That makes total sense, it looks fine to me as is.

I have just tested it myself and can see that unaligned references is already a hard error anyway:

> warning: lint `unaligned_references` has been removed: converted into hard error, see issue #82523 <https://github.com/rust-lang/rust/issues/82523> for more information

So I will leave this as is and close out this PR without merge.

**Pull request closed without merging**